feature: Text_to_Python

idkwhttsay opened the pull request:
## PR Checklist

### Model Interaction

- [x] **Flexible LLM platform support** The platform should be easily switchable. Use LangChain or LlamaIndex.
- [x] **Use prompt guide corresponding to the model**  For example for [Granite 3.x Language Models](https://www.ibm.com/granite/docs/models/granite/#prompt-anatomy)

### Data

- [x] **Example data**: Follow the example data guidance.

### Notebook requirements

- [x] **Notebook outputs cleared**: Ensure all notebook outputs are cleared.
- [x] **Automated testing**: Add the recipe to the automated tests as described [here](https://github.com/ibm-granite-community/community/wiki/Automated-Validation)
- [x] **Test in Google Colab**:
    - [x] Test that it works in Google Colab (Python 3.10.12).
    - [x] Colab has its own package set and Python version, so ensure compatibility.
- [x] **Test locally**:
    - [x] Ensure the code works in a fresh Python virtual environment (venv).
- [x] **Standard access to secrets and variables** Include `!pip install git+https://github.com/ibm-granite-community/utils` in the first code cell in order to make `get_env_var` available to accessing secrets and variables in the recipe.

### Incoming References

- [x] **README.md updates**:
    - [x] Add a link to the recipe in the Table of Contents (ToC).
    - [x] Include a Colab button after that link.

### GitHub

- [x] **Commits signed**: All commits must be GPG or SSH signed.
- [x] **DCO Compliance**: Developer Certificate of Origin (DCO) applies to the code, documentation, and any example data provided. Ensure commits are signed off.

idkwhttsay commented:
Issue #171 

@bjhargrave this PR is incomplete! I've created it for you to look at the midway progress so I can make changes on the go. I'll surely complete all the requirements from the checklist.

For now, can you please look at the basic Text_To_Python recipes?

idkwhttsay commented:
@bjhargrave

bjhargrave commented on recipes/Text_to_Python_FIM/Text_to_Python_FIM.ipynb:
```diff
@@ -0,0 +1,656 @@
+    "from ibm_granite_community.notebook_utils import get_env_var\n",
+    "\n",
+    "model = Replicate(\n",
+    "    model=\"ibm-granite/granite-8b-code-instruct-128k\",\n",
```
There is no code instruct for 3.3. Just instruct: https://replicate.com/ibm-granite/granite-3.3-8b-instruct

So use that model.

idkwhttsay replied:
Okay!

bjhargrave commented on .github/notebook_lists/vanilla_notebooks.txt:
```diff
@@ -5,3 +5,5 @@ recipes/Text_to_Shell_Exec/Text_to_Shell_Exec.ipynb
 recipes/Text_to_SQL/Text_to_SQL.ipynb
 recipes/Unit_Tests_Generation/Unit_Tests_Generation.ipynb
 recipes/Code_Gen_from_Tests/Code_Gen_from_Tests.ipynb
+recipes/Text_to_Python/Text_to_Python.ipynb
+recipes/Text_to_Python_FIM/Text_to_Python_FIM.ipynb
```
You need to fix the path here since it still references the other folder.

idkwhttsay replied:
fixed

idkwhttsay commented:
@bjhargrave now tests should pass, could you run them again?

bjhargrave commented:
You forgot to run pre-commit.

idkwhttsay commented:
> You forgot to run pre-commit.

done

bjhargrave commented:
@idkwhttsay Where are we on finishing up this PR? Thanks.

idkwhttsay commented:
@bjhargrave 
1) passed pre-commit
2) tested in google colab with "ibm-granite/granite-3.3-2b-instruct" and everything worked perfectly. 2b because I didn't have enough memory on a colab VM for an 8b model

idkwhttsay commented on recipes/Text_to_Python/Text_to_Python_FIM.ipynb:
```diff
@@ -36,15 +36,15 @@
     "    langchain_community \\\n",
     "    transformers \\\n",
-    "    replicate"
+    "    torch"
```
do you mean I should use a Replicate hosted model instead of a locally hosted one?

idkwhttsay commented:
@bjhargrave 
1) Recipe is now using replicate and I also added the system prompt template back
2) I decided to stay with python and changed all the examples to Python

bjhargrave commented:
> is there any way to add a replicate api token to GitHub secrets so that every time the test workflow is being run, it doesn't fail because these is no "replicate_api_token"?

No. This is a security feature where PRs from forks do not have access to the main repo's secrets to avoid secret exfiltration by untrusted forks.

bjhargrave commented:
I rebased and squashed the commits. I am also moving the FIM recipe to a separate PR.

bjhargrave commented:
> I am also moving the FIM recipe to a separate PR.

FIM recipes is now in #180
